File: two_pointers/1_two_sum.py
def two_sum(self, nums: list[int], target: int) -> list[int]:
    hash_map = {}
    for i in range(len(nums)):
        needed = target - nums[i]
        if needed in hash_map:
            return [hash_map[needed], i]
        else:
            hash_map[nums[i]] = i

File: two_pointers/1_two_sum.go
func twoSum(nums []int, target int) []int {
	hashMap := make(map[int]int)

	for idx, n := range nums {
		if prevIdx, ok := hashMap[target-n]; ok {
			return []int{prevIdx, idx}
		}
		hashMap[n] = idx
	}

	return nil
}
